hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MetricsCollectionCycle.java

```diff
@@ -101,9 +102,7 @@ void collectDynamicMetrics(Collection<DynamicMetricsProvider> metricsSources) {
     }
 
     void notifyAllGauges(Collection<AbstractGauge> gauges) {
-        for (AbstractGauge gauge : gauges) {
-            gauge.onCollectionCompleted(collectionId);
-        }
+        gauges.stream().filter(Objects::nonNull).forEach(gauge -> gauge.onCollectionCompleted(collectionId));
```

Comment:
@JackPGreen looks fine for the logic. I would move the `null` check within the existing `for` to avoid the extra cost of the `map` and `stream` at the cost of aesthetics. The metrics system is hit a lot and allocates a lot (proportionally) of memory in very short and frequent cycles.

@vbekiaris ?